source-reminder/lambda_function.py

```diff
@@ -0,0 +1,146 @@
+import gspread
+from oauth2client.service_account import ServiceAccountCredentials
+import requests
+from datetime import date, datetime, timedelta
+import logging
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+scope = ['https://spreadsheets.google.com/feeds', 'https://www.googleapis.com/auth/drive', 'https://www.googleapis.com/auth/admin.directory.user']
+API_ENDPOINT = 'https://www.texastribune.org/api/v2/articles/'
+credentials = ServiceAccountCredentials.from_json_keyfile_name('credentials.json', scope)  # Add credentials to the service account.
+client = gspread.authorize(credentials)  # Authorize the client sheet.
+sheet = client.open('Source Diversity Submission (Responses) 2021')  # Get the instance of the spreadsheet.
+sheet_instance = sheet.get_worksheet(1)  # Get instance of the second sheet.
+records_data = sheet_instance.get_all_records()  # Get all records of the data of the instance.
+empty_line = {'Timestamp': '', 'Your Name': '', 'Headline ': '', 'Link': '',
+              'Date of Publication': '', 'Is this for the ProPublica/The Texas Tribune investigative unit?': '',
+              'Type': '', 'Topic': '', 'Byline': '', 'Name of Source': '', 'Type of Source (choose best option)': '',
+              'Is this story based on news made by this person?': '',
+              'Did you confirm with this source how they identify in terms of race and gender?': '',
+              'Any notes about how asking this source went?': '',
+              'Gender': '', 'Race/Ethnicity': '', 'Do you have another source? ': '', 'Email Address': ''}
+records_data.remove(empty_line)  # Edge case.
+start = date.today() - timedelta(weeks=3)
+end = date.today() - timedelta(weeks=1)
+
+# View the data for the relevant window of time.
+timeframe_records = sorted(
+    filter(lambda x: start <= datetime.strptime(x['Date of Publication'], '%m/%d/%Y').date() <= end, records_data),
+    key=lambda i: i['Date of Publication'])
+sheet_headlines = [k['Headline '] for k in timeframe_records]
+
+
+def request_data(limit, start_date, end_date):
+    query = {'limit': limit, 'start_date': start_date, 'end_date': end_date}
+    return requests.get(API_ENDPOINT, params=query)
```

Comment:
Maybe overkill, but might be good to catch any errors that resulted from this request

Reply:
Agree with Ashley about the hard-coded field names above & the helpfulness of error catching here.   For Hoedown Helper slackbot we had a couple instances where a column name in Airtable got changed and it failed kinda silently, if there's not a good way to avoid the dependency on the specific Sheet and column names, and also for any other random errors that may occur (API unavailable, etc.) it would be good to catch that.

We could also look into connecting this to Sentry so errors get sent to Slack.